Routing.md

```diff
@@ -0,0 +1,59 @@
+# Routing Meatdata Payload Extension
+
+## Introduction
+Specify a metadata payload for [RSocket](https://rsocket.io) that contains both information about where to route a [`PAYLOAD` frame](https://github.com/rsocket/rsocket/blob/master/Protocol.md#frame-payload) as well as metadata for the frame itself.
+
+## Metadata Payload
+The metadata payload has a MIME Type of `message/x.rsocket.routing`.
+
+### Payload Contents
+```
+     0                   1                   2                   3
+     0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+    +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+    |            Version            |T|P|S|D|   Flags   |
+    +-------------------------------+-+-+-+-+-----------+-----------+
+    |      Route Token Length       |   Route Token                ...
+    +-------------------------------+-------------------------------+
+    |                                                               |
+    +                                                               +
+    |                                                               |
+    +                           Trace ID                            +
+    |                                                               |
+    +                                                               +
+    |                                                               |
+    +---------------------------------------------------------------+
+    |                                                               |
+    +                           Span ID                             +
+    |                                                               |
+    +---------------------------------------------------------------+
+    |                                                               |
+    +                        Parent Span ID                         +
+    |                                                               |
+    +-+-+-----------------------------------------------------------+
+    |S|D|
+    +-+-+-----------+-----------------------------------------------+
+    |  MIME Length  |   Metadata Encoding MIME Type                ...
+    +---------------+-----------------------------------------------+
+    |  MIME Length  |   Data Encoding MIME Type                    ...
+    +---------------+-----------------------------------------------+
+                          Metadata Payload
+```
+
+* **Version**: (16 bits = max value 65,535) Unsigned 16-bit integer of version number of the routing payload
+* **Flags**: (10 bits)
```

Comment:
Isn’t byte aligning easier? E.g. padding out to byte and not making things optional at bit level

Reply:
Even just for sanity of people reading the hex stream.